packages/block-library/src/gallery/deprecated.js

```diff
@@ -15,6 +15,139 @@ import { RichText } from '@wordpress/block-editor';
 import { defaultColumnsNumber } from './shared';
 
 const deprecated = [
+	{
+		attributes: {
+			images: {
+				type: 'array',
+				default: [],
+				source: 'query',
+				selector: '.blocks-gallery-item',
+				query: {
+					url: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'src',
+					},
+					fullUrl: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-full-url',
+					},
+					link: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-link',
+					},
+					alt: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'alt',
+						default: '',
+					},
+					id: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-id',
+					},
+					caption: {
+						type: 'string',
+						source: 'html',
+						selector: '.blocks-gallery-item__caption',
+					},
+				},
+			},
+			ids: {
+				type: 'array',
+				default: [],
+			},
+			columns: {
+				type: 'number',
+			},
+			caption: {
+				type: 'string',
+				source: 'html',
+				selector: '.blocks-gallery-caption',
+			},
+			imageCrop: {
+				type: 'boolean',
+				default: true,
+			},
+			linkTo: {
+				type: 'string',
+				default: 'none',
+			},
+		},
+		isEligible( { ids } ) {
+			return ids &&
+				ids.length > 0 &&
+				ids.reduce( ( state, id ) => {
+					return state || ( typeof id === 'string' );
+				}, false );
+		},
+		migrate( attributes ) {
+			return {
+				...attributes,
+				ids: map( attributes.ids, ( id ) => {
+					if ( ! id ) {
+						return null;
+					}
+					if ( typeof id === 'string' ) {
+						return parseInt( id, 10 );
+					}
+					if ( typeof id === 'number' ) {
+						return id;
+					}
+
+					return null;
+				} ),
```

Comment:
This could be

```js
const parsedId = parseInt( id, 10 );
return Number.isInteger( parsedId ) ? parsedId : null;
```

Reply:
@mcsf  Will this work if the original id was a number?

Reply:
Yes:

> The value to parse. If the string argument is not a string, then it is converted to a string (using the ToString abstract operation). Leading whitespace in the string argument is ignored.
— [parseInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt)

Reply:
Thanks for the clarification, implemented in: https://github.com/WordPress/gutenberg/pull/19163/commits/b40dca113bbf205dc6409a3b6679d8dd38e5699e